citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/CreateAuthorityController.java

```diff
@@ -0,0 +1,79 @@
+package edu.asu.diging.citesphere.web.user.authorities;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.citesphere.core.service.IAuthorityService;
+import edu.asu.diging.citesphere.core.service.ICitationManager;
+import edu.asu.diging.citesphere.model.authority.IAuthorityEntry;
+import edu.asu.diging.citesphere.model.authority.impl.AuthorityEntry;
+import edu.asu.diging.citesphere.user.IUser;
+import edu.asu.diging.citesphere.web.forms.CreateAuthorityForm;
+
+@Controller
+public class CreateAuthorityController {
+
+    @Autowired
+    private IAuthorityService authorityService;
+
+    @Autowired
+    private ICitationManager citationManager;
+
+    @GetMapping(value = { "/auth/authority/create", "/auth/authority/{zoteroGroupId}/create" })
+    public String show(Authentication authentication, Model model,
+            @PathVariable(required = false, value = "zoteroGroupId") String zoteroGroupId) {
+        IUser user = (IUser) authentication.getPrincipal();
+        model.addAttribute("groups", citationManager.getGroups(user));
+        CreateAuthorityForm authorityForm = new CreateAuthorityForm();
+        if (zoteroGroupId != null) {
```

Comment:
you can probably skip this. If `zoteroGroupId` is null, it will just replace null with null.